File: src/main/java/org/micoli/php/configuration/schema/valueGenerator/ActionIdValueGenerator.java
package org.micoli.php.configuration.schema.valueGenerator;

import com.intellij.openapi.actionSystem.ActionManager;
import java.util.List;

public class ActionIdValueGenerator implements PropertyValueGenerator {
    List<String> fieldNames = List.of("actionId");

    @Override
    public ReferenceType getType() {
        return ReferenceType.AS_REF;
    }

    @Override
    public String getRefId() {
        return "actionIds";
    }

    @Override
    public List<String> getFieldNames() {
        return fieldNames;
    }

    @Override
    public List<String> getValues() {
        return ActionManager.getInstance().getActionIdList("").stream()
                .filter(s -> !s.contains("anonymous-group-"))
                .distinct()
                .sorted()
                .toList();
    }
}

File: src/main/kotlin/org/micoli/php/configuration/schema/valueGenerator/ActionIdValueGenerator.kt
package org.micoli.php.configuration.schema.valueGenerator

import com.intellij.openapi.actionSystem.ActionManager
import kotlinx.collections.immutable.ImmutableList
import kotlinx.collections.immutable.persistentListOf
import kotlinx.collections.immutable.toImmutableList

class ActionIdValueGenerator : PropertyValueGenerator {
    override fun getFieldNames(): ImmutableList<String> = persistentListOf("actionId")

    override fun getValues(): ImmutableList<String> =
      ActionManager.getInstance()
        .getActionIdList("")
        .stream()
        .filter { s: String? -> s != null }
        .filter { s: String -> s.contains("anonymous-group-") }
        .distinct()
        .sorted()
        .toList()
        .toImmutableList()
}
